nomad/state/state_store_host_volumes.go

```diff
@@ -290,3 +290,21 @@ func upsertHostVolumeForNode(txn *txn, node *structs.Node, index uint64) error {
 
 	return nil
 }
+
+func (s *StateStore) NodeHasHostVolume(nodeID, volName string) bool {
+	iter, err := s.HostVolumesByNodeID(nil, nodeID, SortDefault)
+	if err != nil {
+		return false
```

Comment:
might we want to err (heh) on the side of caution and say "yes, do not put another volume here" (`return true`), since state is in some kind of terrible trouble?

Reply:
This error value is part of a larger problem which is that the _only_ way memdb returns errors from queries is if you ask it for a table or index that doesn't exist. Which should panic! We don't accept random table names from our users here, so if the table or index doesn't exist it's a programmer error. But we've hesitated to implement that across the board (probably for fear someone would add a different possible error case to memdb without letting us know).

All of which is to say, yeah, agreed. Done. :grin: